src/components/ProductList.vue

```diff
@@ -45,12 +46,18 @@ export default {
       ],
       showForm: false,
       productIndex: '',
-      showDeleteDialog: false
+      showDeleteDialog: false,
+      showMessage: false
     }
   },
   mounted () {
     this.$store.dispatch('products/ShowDataInTable')
   },
+  watch: {
+    '$store.state.products.products': function (newValue) {
+      if (newValue.length) { this.$store.dispatch('products/ShowDataInTable') } else { this.showMessage = true }
+    }
```

Comment:
I didn't understand the code you are using here to watch something, Try to refer vue documentation if you got some doubt regarding the usage.

You have already got your products in computed property, So if you want to watch that state,
*watch: {
product (newValue, oldValue) {
Do what ever you want here, if got a check use if  etc
}
},*